src/main/java/com/epam/grid/engine/provider/job/sge/SgeJobProvider.java

```diff
@@ -161,19 +159,48 @@ public CommandType getProviderType() {
     @Override
     public Job runJob(final JobOptions options) {
         validateJobOptions(options);
-        return buildNewJob(getResultOfExecutedCommand(simpleCmdExecutor, makeQsubCommand(options)));
+        return buildNewJob(toSubmitNewJob(options));
+    }
+
+    private long toSubmitNewJob(final JobOptions options) {
+        final CommandResult result = simpleCmdExecutor.execute(makeQsubCommand(options));
+        if (result.getExitCode() != 0) {
+            CommandsUtils.throwExecutionDetails(result, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        final Matcher matcher = SUBMITTED_JOB_ID_PATTERN.matcher(result.getStdOut().get(0));
```

Comment:
possible NPE here, when `get(0)` is called

Reply:
fixed the if-condition